go.mod: patch with lib/pq TimeTZ fix

otan opened the pull request:
Resolves #65066.

See individual commits for details.

otan committed:
go.mod: patch with lib/pq TimeTZ fix

Apply a custom patch of TimeTZ on the cockroachdb fork of lib/pq to fix
a TimeTZ issue. We cannot bump lib/pq to the latest version as there is
some sort of mismatch with the pgwire protocol introduced on lib/pq.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65321)
<!-- Reviewable:end -->

rafiss commented on DEPS.bzl:
```diff
@@ -2429,8 +2429,9 @@ def go_deps():
         importpath = "github.com/lib/pq",
-        sum = "h1:9xohqzkUwzR4Ga4ivdTcawVS89YSDVxXMa3xJX3cGzg=",
-        version = "v1.8.0",
+        replace = "github.com/cockroachdb/pq",
```
https://github.com/lib/pq/pull/1039 was merged so should we just tag/release that?

otan replied:
see the comment in go.mod below -- we can't use lib/pq on master because of #65320 ...

rafiss commented on pkg/sql/logictest/testdata/logic_test/timetz:
```diff
@@ -670,3 +670,8 @@ SELECT a FROM regression_44774 ORDER BY a
 DROP TABLE regression_44774
+
+query T
+SELECT '20:01:02+03:04:05' AS regression_65066
```
does this test catch it? i thought the issue was in the CLI so we'd need something in cli_test.go?

(when i connect to crdb using psql, this query works fine on master. do logic tests use the cli formatting?)

rafiss commented on pkg/cli/cli_test.go:
```diff
@@ -196,6 +196,7 @@ func Example_sql() {
 	// and with the query below the division by zero error will occur after the
 	// first batch consisting of 1 row has been returned to the client.
 	c.RunWithArgs([]string{`sql`, `-e`, `select 1/(@1-2) from generate_series(1,3)`})
+	c.RunWithArgs([]string{`sql`, `-e`, `SELECT '20:01:02+03:04:05' AS regression_65066`})
```
oops sorry i missed before, but we'd need the `::timetz` cast here right? otherwise it is just a string value.

otan replied:
![my bad](https://user-images.githubusercontent.com/3646147/118586263-c83c9100-b7dd-11eb-86b3-6ca43de2ac22.png)

rafiss reviewed (changes requested):
one more issue, then lgtm -- though as i mentioned i'll poke around at the other issue that's blocking the lib/pq upgrade first

rafiss reviewed (approved):
lgtm! let's merge and i hope i can come back and look at https://github.com/cockroachdb/cockroach/issues/65320 later

> Reviewed 3 of 4 files at r1, 1 of 2 files at r3, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65321#-:-MadwKxb-O5Bhu5BPeEC:bhnts1x)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=rafiss

otan commented:
bors r-

otan committed:
logictest/timetz: add regression test for second offset TimeTZ

Release note (bug fix): Fixed a bug where TimeTZ values would not
display appopriately in the CLI.

otan commented:
bors r=rafiss

otan commented:
bors r=rafiss
